corehq/apps/es/management/commands/resave_failed_forms_and_cases.py

```diff
@@ -14,52 +19,83 @@
 from corehq.form_processor.utils.xform import resave_form
 from corehq.apps.hqcase.utils import resave_case
 from corehq.util.log import with_progress_bar
+from dimagi.utils.chunked import chunked
+
+DATE_FORMAT = "%Y-%m-%d"
+
+
+def valid_date(s):
+    try:
+        return datetime.strptime(s, DATE_FORMAT)
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
 
 
 class Command(BaseCommand):
     def add_arguments(self, parser):
         parser.add_argument('domain')
+        parser.add_argument(
+            'start_date',
+            type=valid_date,
```

Comment:
can you use `type=datetime.date` here instead of a custom parser

Reply:
Honestly I didn't really think to hard about it and just copied it from here: https://github.com/dimagi/commcare-hq/pull/20999/files#diff-972d6eb693aff7ef3de866a488625cdeR24

I meant to extract this into a utility since it's used in 3 places now